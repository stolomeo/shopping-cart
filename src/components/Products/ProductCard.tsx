import { Grid } from "@mui/material";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import { Product } from "../../types";

type Props = {
  product: Product;
};

export default function ProductCard({ product }: Props) {
  const { title, price, description, category, image, rating } = product;

  return (
    <Grid item xs={4}>
      <Card>
        <CardMedia component="img" image={image} alt="product" />
        <CardContent>
          <Typography variant="h6" color="text.secondary" fontWeight="bold">
            {title}
          </Typography>
          <Typography variant="body1" color="text.secondary" fontWeight="bold">
            {price}
          </Typography>
        </CardContent>
        <CardActions sx={{ px: 2 }}>
          <Button fullWidth={true} variant="contained" color="primary">
            Add To Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
}
