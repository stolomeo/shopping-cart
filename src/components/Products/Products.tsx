import { Container, Grid } from "@mui/material";
import { useContext, useEffect, useState } from "react";
import CartCountContext from "../../context/CartCountContext";
import { Product } from "../../types";
import { getData } from "../../utils";
import ProductCard from "./ProductCard/ProductCard";

export default function Products() {
  const [products, setProducts] = useState<Product[]>();
  const [cartitems, setCartitems] = useState<Product[]>();
  const { cartCount, setCartCount } = useContext(CartCountContext);

  const handleAddToCart = (id: number) => {
    setCartCount(cartCount + 1);
  };

  useEffect(() => {
    (async () => {
      setProducts(await getData(9));
    })();
  }, []);

  const productElements = !products ? (
    <h1>Loading...</h1>
  ) : (
    products.map((product) => {
      const { id } = product as Product;
      return (
        <ProductCard key={id} {...product} handleAddToCart={handleAddToCart} />
      );
    })
  );

  return (
    <Container maxWidth="lg" sx={{ mt: 3 }}>
      <Grid container spacing={4}>
        {productElements}
      </Grid>
    </Container>
  );
}
