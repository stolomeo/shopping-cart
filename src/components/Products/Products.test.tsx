import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Products from "./Products";

describe("<Products />", () => {
  const renderComponent = () => render(<Products />);

  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should demonstrate this matcher's usage with react testing library", async () => {
    const { container } = renderComponent();
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
