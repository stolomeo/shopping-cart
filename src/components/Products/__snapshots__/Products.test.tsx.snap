// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-rd586k-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
    >
      <h1>
        Loading...
      </h1>
    </div>
  </div>
</div>
`;
